<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Data Acquisition and Preparation | NAME OF WORKSHOP YEAR</title>
  <meta name="description" content="Workshop website for the YEAR WORKSHOP Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Data Acquisition and Preparation | NAME OF WORKSHOP YEAR" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://cbw-dev.github.io/bookdown-template/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the YEAR WORKSHOP Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="cbw-dev/bookdown-template" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Data Acquisition and Preparation | NAME OF WORKSHOP YEAR" />
  
  <meta name="twitter:description" content="Workshop website for the YEAR WORKSHOP Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://cbw-dev.github.io/bookdown-template/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: INSTRUCTOR AND TA NAMES" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"/>
<link rel="next" href="effect-size-calculation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Pharmacogenomics Data Analysis<br><br></a></strong></center></li>
<a href="./">
 <img src="img/pgx.png" width="70%" alt="Pharmacogenomics Data Analysis logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-and-compute-setup.html"><a href="data-and-compute-setup.html"><i class="fa fa-check"></i>Data and Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><i class="fa fa-check"></i>Module 1: Intro to Pharmacogenomics in the Context of Cancer</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#lab"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html"><i class="fa fa-check"></i>Module 2: Data Preparation for Pharmacogenomic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#lab-1"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><i class="fa fa-check"></i>Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lab-2"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><i class="fa fa-check"></i>Module 4: Pharmacogenomics for Biomarker Discovery - Advanced Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#lab-3"><i class="fa fa-check"></i>Lab</a></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#instructors-names-and-contact-information"><i class="fa fa-check"></i>Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#introduction"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#learning-objectives"><i class="fa fa-check"></i>Learning Objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-acquisition-and-preparation.html"><a href="data-acquisition-and-preparation.html"><i class="fa fa-check"></i>Data Acquisition and Preparation</a>
<ul>
<li class="chapter" data-level="" data-path="data-acquisition-and-preparation.html"><a href="data-acquisition-and-preparation.html#extracting-required-vectors-for-biomarker-analysis"><i class="fa fa-check"></i>Extracting required vectors for biomarker analysis</a>
<ul>
<li class="chapter" data-level="" data-path="data-acquisition-and-preparation.html"><a href="data-acquisition-and-preparation.html#gene-expression-data"><i class="fa fa-check"></i>Gene Expression Data</a></li>
<li class="chapter" data-level="" data-path="data-acquisition-and-preparation.html"><a href="data-acquisition-and-preparation.html#drug-response-data"><i class="fa fa-check"></i>Drug Response Data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-acquisition-and-preparation.html"><a href="data-acquisition-and-preparation.html#gene-expression-exploratory-data-analysis-eda"><i class="fa fa-check"></i>Gene Expression Exploratory Data Analysis (EDA)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="effect-size-calculation.html"><a href="effect-size-calculation.html"><i class="fa fa-check"></i>Effect Size Calculation</a></li>
<li class="chapter" data-level="" data-path="perform-meta-analysis.html"><a href="perform-meta-analysis.html"><i class="fa fa-check"></i>Perform Meta-Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="perform-meta-analysis.html"><a href="perform-meta-analysis.html#visualization-with-forest-plots"><i class="fa fa-check"></i>Visualization with Forest Plots</a>
<ul>
<li class="chapter" data-level="" data-path="perform-meta-analysis.html"><a href="perform-meta-analysis.html#forest-function-from-meta-package"><i class="fa fa-check"></i>forest function from meta package</a></li>
<li class="chapter" data-level="" data-path="perform-meta-analysis.html"><a href="perform-meta-analysis.html#using-ggplot2"><i class="fa fa-check"></i>Using ggplot2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html"><i class="fa fa-check"></i>Subgroup Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html#data-preparation"><i class="fa fa-check"></i>Data Preparation</a></li>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html#statistical-analysis-to-measure-drug-response-associations"><i class="fa fa-check"></i>Statistical Analysis to measure Drug Response Associations</a>
<ul>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html#function-for-computing-concordance-index"><i class="fa fa-check"></i>Function for computing concordance index</a></li>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html#visualizing-feature-drug-associations"><i class="fa fa-check"></i>Visualizing feature-drug associations</a></li>
<li class="chapter" data-level="" data-path="subgroup-analysis.html"><a href="subgroup-analysis.html#identifying-candidate-biomarkers"><i class="fa fa-check"></i>Identifying candidate biomarkers</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="interactive-exercise.html"><a href="interactive-exercise.html"><i class="fa fa-check"></i>Interactive Exercise</a></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">

<br><br>

<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="70" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">

<br><br>

<a href="https://computationalgenomics.ca/"><img src="img/sponsors/c3g.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Canadian Centre for Computational Genomics logo.">

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">NAME OF WORKSHOP YEAR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-acquisition-and-preparation" class="section level1 hasAnchor">
<h1>Data Acquisition and Preparation<a href="data-acquisition-and-preparation.html#data-acquisition-and-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Like before, we will use smaller versions of the PharmacoSets for the workshop to minimize
processing time. Let’s load these datasets in and subset for common cell lines and .</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-acquisition-and-preparation.html#cb4-1" tabindex="-1"></a><span class="co"># load in the datasets</span></span>
<span id="cb4-2"><a href="data-acquisition-and-preparation.html#cb4-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GDSCsmall&quot;</span>)</span>
<span id="cb4-3"><a href="data-acquisition-and-preparation.html#cb4-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;CCLEsmall&quot;</span>)</span></code></pre></div>
<div id="extracting-required-vectors-for-biomarker-analysis" class="section level2 hasAnchor">
<h2>Extracting required vectors for biomarker analysis<a href="data-acquisition-and-preparation.html#extracting-required-vectors-for-biomarker-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Recall there are two components needed for a biomarker association analysis:
- Feature data (molecular profile)
- Drug response</p>
<p>Say we are interested in <code>ENSG00000003987</code> expression association with <code>PD-0325901</code>.
We can start by extracting this data from both PSets.</p>
<div id="gene-expression-data" class="section level3 hasAnchor">
<h3>Gene Expression Data<a href="data-acquisition-and-preparation.html#gene-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="data-acquisition-and-preparation.html#cb5-1" tabindex="-1"></a><span class="co"># specify our gene of interest</span></span>
<span id="cb5-2"><a href="data-acquisition-and-preparation.html#cb5-2" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;ENSG00000003987&quot;</span></span>
<span id="cb5-3"><a href="data-acquisition-and-preparation.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="data-acquisition-and-preparation.html#cb5-4" tabindex="-1"></a></span>
<span id="cb5-5"><a href="data-acquisition-and-preparation.html#cb5-5" tabindex="-1"></a><span class="co"># extract the RNA-Seq data for our gene of interest in both GDSC and CCLE</span></span>
<span id="cb5-6"><a href="data-acquisition-and-preparation.html#cb5-6" tabindex="-1"></a>ccle_rna_data <span class="ot">&lt;-</span> <span class="fu">summarizeMolecularProfiles</span>(</span>
<span id="cb5-7"><a href="data-acquisition-and-preparation.html#cb5-7" tabindex="-1"></a>  CCLEsmall,</span>
<span id="cb5-8"><a href="data-acquisition-and-preparation.html#cb5-8" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;rna&quot;</span>, <span class="at">features =</span> gene</span>
<span id="cb5-9"><a href="data-acquisition-and-preparation.html#cb5-9" tabindex="-1"></a>) <span class="sc">|&gt;</span> <span class="fu">assay</span>()</span>
<span id="cb5-10"><a href="data-acquisition-and-preparation.html#cb5-10" tabindex="-1"></a></span>
<span id="cb5-11"><a href="data-acquisition-and-preparation.html#cb5-11" tabindex="-1"></a>gdsc_rna_data <span class="ot">&lt;-</span> <span class="fu">summarizeMolecularProfiles</span>(</span>
<span id="cb5-12"><a href="data-acquisition-and-preparation.html#cb5-12" tabindex="-1"></a>  GDSCsmall,</span>
<span id="cb5-13"><a href="data-acquisition-and-preparation.html#cb5-13" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;rna&quot;</span>, <span class="at">features =</span> gene</span>
<span id="cb5-14"><a href="data-acquisition-and-preparation.html#cb5-14" tabindex="-1"></a>) <span class="sc">|&gt;</span> <span class="fu">assay</span>()</span></code></pre></div>
<pre><code>##   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="data-acquisition-and-preparation.html#cb7-1" tabindex="-1"></a>gdsc_rna_data</span></code></pre></div>
<pre><code>##                    22RV1 23132-87      380     5637    639-V    647-V     697
## ENSG00000003987 4.197327 4.022529 4.000509 4.031896 3.977781 3.834624 4.36307
##                    769-P    786-0  8-MG-BA
## ENSG00000003987 4.397028 3.973001 4.275385</code></pre>
</div>
<div id="drug-response-data" class="section level3 hasAnchor">
<h3>Drug Response Data<a href="data-acquisition-and-preparation.html#drug-response-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, let’s take a look at the available drug response data</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="data-acquisition-and-preparation.html#cb9-1" tabindex="-1"></a><span class="fu">sensitivityProfiles</span>(CCLEsmall) <span class="sc">|&gt;</span> <span class="fu">head</span>()</span></code></pre></div>
<pre><code>##                        ic50_published auc_published amax_published
## drugid_AEW541_1321N1         8.000000     0.0873750      -42.55801
## drugid_AEW541_22Rv1          2.329924     0.2205000      -71.58934
## drugid_AEW541_42-MG-BA       2.682130     0.1144375      -63.49137
## drugid_AEW541_5637           5.002314     0.1243550      -62.35278
## drugid_AEW541_639-V          1.736181     0.1936250      -51.95981
## drugid_AEW541_697            4.260822     0.1087625      -73.33379
##                        auc_recomputed ic50_recomputed amax_recomputed
## drugid_AEW541_1321N1        0.1017047        13.09700        42.87563
## drugid_AEW541_22Rv1         0.2239774         2.18654        70.37213
## drugid_AEW541_42-MG-BA      0.1215999         2.68981        63.00413
## drugid_AEW541_5637          0.1092448         5.19939        61.43152
## drugid_AEW541_639-V         0.1912847         5.40523        54.22604
## drugid_AEW541_697           0.1146054         4.17688        69.94055</code></pre>
<p>We can see a few options of drug response metrics to choose from. We will use
<code>auc_published</code> for our drug response associations.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data-acquisition-and-preparation.html#cb11-1" tabindex="-1"></a><span class="co"># extract AUC_published data from CCLE and GDSC</span></span>
<span id="cb11-2"><a href="data-acquisition-and-preparation.html#cb11-2" tabindex="-1"></a>ccle_auc <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb11-3"><a href="data-acquisition-and-preparation.html#cb11-3" tabindex="-1"></a>  CCLEsmall,</span>
<span id="cb11-4"><a href="data-acquisition-and-preparation.html#cb11-4" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb11-5"><a href="data-acquisition-and-preparation.html#cb11-5" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb11-6"><a href="data-acquisition-and-preparation.html#cb11-6" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb11-7"><a href="data-acquisition-and-preparation.html#cb11-7" tabindex="-1"></a>)</span>
<span id="cb11-8"><a href="data-acquisition-and-preparation.html#cb11-8" tabindex="-1"></a></span>
<span id="cb11-9"><a href="data-acquisition-and-preparation.html#cb11-9" tabindex="-1"></a>gdsc_auc <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb11-10"><a href="data-acquisition-and-preparation.html#cb11-10" tabindex="-1"></a>  GDSCsmall,</span>
<span id="cb11-11"><a href="data-acquisition-and-preparation.html#cb11-11" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb11-12"><a href="data-acquisition-and-preparation.html#cb11-12" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb11-13"><a href="data-acquisition-and-preparation.html#cb11-13" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb11-14"><a href="data-acquisition-and-preparation.html#cb11-14" tabindex="-1"></a>)</span>
<span id="cb11-15"><a href="data-acquisition-and-preparation.html#cb11-15" tabindex="-1"></a></span>
<span id="cb11-16"><a href="data-acquisition-and-preparation.html#cb11-16" tabindex="-1"></a>ccle_auc[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##               1321N1 143B    22RV1 23132-87 253J-BV
## PD-0325901 0.1148875   NA 0.385000       NA      NA
## 17-AAG     0.4177000   NA 0.372460       NA      NA
## AEW541     0.0873750   NA 0.220500       NA      NA
## Nilotinib         NA   NA 0.000000       NA      NA
## PHA-665752 0.0333750   NA 0.094375       NA      NA</code></pre>
<p>The <code>summarizeSensitivityProfiles()</code> function nicely pulls the drug response data
and converts it into a sample~drug matrix for easy use.</p>
<p>We only need the drug response data for <code>PD-0325901</code>, so let’s pull that out.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data-acquisition-and-preparation.html#cb13-1" tabindex="-1"></a>drug <span class="ot">&lt;-</span> <span class="st">&quot;PD-0325901&quot;</span></span>
<span id="cb13-2"><a href="data-acquisition-and-preparation.html#cb13-2" tabindex="-1"></a>ccle_drug_data <span class="ot">&lt;-</span> ccle_auc[drug, ]</span>
<span id="cb13-3"><a href="data-acquisition-and-preparation.html#cb13-3" tabindex="-1"></a>gdsc_drug_data <span class="ot">&lt;-</span> gdsc_auc[drug, ]</span>
<span id="cb13-4"><a href="data-acquisition-and-preparation.html#cb13-4" tabindex="-1"></a></span>
<span id="cb13-5"><a href="data-acquisition-and-preparation.html#cb13-5" tabindex="-1"></a><span class="fu">head</span>(ccle_drug_data)</span></code></pre></div>
<pre><code>##    1321N1      143B     22RV1  23132-87   253J-BV      253J 
## 0.1148875        NA 0.3850000        NA        NA        NA</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data-acquisition-and-preparation.html#cb15-1" tabindex="-1"></a><span class="fu">head</span>(gdsc_drug_data)</span></code></pre></div>
<pre><code>##    22RV1 23132-87      380     5637    639-V    647-V 
## 0.029649 0.122911       NA 0.007242 0.279261 0.100351</code></pre>
</div>
</div>
<div id="gene-expression-exploratory-data-analysis-eda" class="section level2 hasAnchor">
<h2>Gene Expression Exploratory Data Analysis (EDA)<a href="data-acquisition-and-preparation.html#gene-expression-exploratory-data-analysis-eda" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before performing the statistical analysis, let’s quickly look at the distribution
of <code>ENSG00000003987</code> expression data.</p>
<p>First, we can check the number of cell lines in each vector:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="data-acquisition-and-preparation.html#cb17-1" tabindex="-1"></a><span class="fu">paste</span>(<span class="st">&quot;Number of cell lines in GDSC:&quot;</span>, <span class="fu">length</span>(gdsc_rna_data))</span></code></pre></div>
<pre><code>## [1] &quot;Number of cell lines in GDSC: 10&quot;</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="data-acquisition-and-preparation.html#cb19-1" tabindex="-1"></a><span class="fu">paste</span>(<span class="st">&quot;Number of cell lines in CCLE:&quot;</span>, <span class="fu">length</span>(ccle_rna_data))</span></code></pre></div>
<pre><code>## [1] &quot;Number of cell lines in CCLE: 1061&quot;</code></pre>
<p>Notice there is a large difference between the number of samples between GDSC and CCLE.
This difference in sample size may reflected in the meta-analysis.</p>
<p>Let’s quickly look at the distribution of this gene in both PSets.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="data-acquisition-and-preparation.html#cb21-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">t</span>(ccle_rna_data), <span class="fu">aes_string</span>(<span class="at">x =</span> gene)) <span class="sc">+</span></span>
<span id="cb21-2"><a href="data-acquisition-and-preparation.html#cb21-2" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">&quot;skyblue&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-3"><a href="data-acquisition-and-preparation.html#cb21-3" tabindex="-1"></a>        <span class="fu">labs</span>(</span>
<span id="cb21-4"><a href="data-acquisition-and-preparation.html#cb21-4" tabindex="-1"></a>          <span class="at">title =</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">&quot;Distribution of &quot;</span>, gene, <span class="st">&quot; Expression in CCLEsmall&quot;</span>),</span>
<span id="cb21-5"><a href="data-acquisition-and-preparation.html#cb21-5" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;Expression&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span></span>
<span id="cb21-6"><a href="data-acquisition-and-preparation.html#cb21-6" tabindex="-1"></a>        )</span>
<span id="cb21-7"><a href="data-acquisition-and-preparation.html#cb21-7" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">t</span>(gdsc_rna_data), <span class="fu">aes_string</span>(<span class="at">x =</span> gene)) <span class="sc">+</span></span>
<span id="cb21-8"><a href="data-acquisition-and-preparation.html#cb21-8" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>, <span class="at">fill =</span> <span class="st">&quot;skyblue&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-9"><a href="data-acquisition-and-preparation.html#cb21-9" tabindex="-1"></a>        <span class="fu">labs</span>(</span>
<span id="cb21-10"><a href="data-acquisition-and-preparation.html#cb21-10" tabindex="-1"></a>          <span class="at">title =</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">&quot;Distribution of &quot;</span>, gene, <span class="st">&quot; Expression in GDSCsmall&quot;</span>),</span>
<span id="cb21-11"><a href="data-acquisition-and-preparation.html#cb21-11" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;Expression&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span></span>
<span id="cb21-12"><a href="data-acquisition-and-preparation.html#cb21-12" tabindex="-1"></a>        )</span>
<span id="cb21-13"><a href="data-acquisition-and-preparation.html#cb21-13" tabindex="-1"></a><span class="fu">ggarrange</span>(p1, p2, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_EDA-1.png" width="672" /></p>
<p>Notice that the difference in distribution shapes. You can imagine what implications
a small sample size (e.g. GDSC) would have when downstream analysis.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="effect-size-calculation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/cbw-dev/PGX-2024/blob/main/040-module-4.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
