<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis | Pharmacogenomics Data Analysis 2024</title>
  <meta name="description" content="Workshop website for the 2024 Pharmacogenomics Data Analysis Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis | Pharmacogenomics Data Analysis 2024" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/PGX_2024/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2024 Pharmacogenomics Data Analysis Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="bioinformaticsdotca/PGX_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis | Pharmacogenomics Data Analysis 2024" />
  
  <meta name="twitter:description" content="Workshop website for the 2024 Pharmacogenomics Data Analysis Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/PGX_2024/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: Benjamin Haibe-Kains, Jermiah Joseph, Nikta Feizi, Julia Nguyen, and Nia Hughes" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-2-data-preparation-for-pharmacogenomic-analysis.html"/>
<link rel="next" href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Pharmacogenomics Data Analysis<br><br></a></strong></center></li>
<a href="./">
 <img src="img/pgx.png" width="70%" alt="Pharmacogenomics Data Analysis logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#class-photo"><i class="fa fa-check"></i>Class Photo</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-and-compute-setup.html"><a href="data-and-compute-setup.html"><i class="fa fa-check"></i>Data and Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><i class="fa fa-check"></i>Module 1: Intro to Pharmacogenomics in the Context of Cancer</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#lab"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#overview"><i class="fa fa-check"></i>Overview</a></li>
<li class="chapter" data-level="" data-path="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html"><a href="module-1-intro-to-pharmacogenomics-in-the-context-of-cancer.html#getting-started"><i class="fa fa-check"></i>Getting Started</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html"><i class="fa fa-check"></i>Module 2: Data Preparation for Pharmacogenomic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#overview-1"><i class="fa fa-check"></i>Overview</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#setup"><i class="fa fa-check"></i>Setup</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#understanding-the-pharmacoset-data-structure"><i class="fa fa-check"></i>Understanding the <code>PharmacoSet</code> Data Structure</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#molecular-profiles-1"><i class="fa fa-check"></i>Molecular Profiles</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#treatment-response"><i class="fa fa-check"></i>Treatment Response</a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#subsetting-a-pharmacoset"><i class="fa fa-check"></i>Subsetting a <code>PharmacoSet</code></a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#computing-over-a-treatmentresponseexperiment"><i class="fa fa-check"></i>Computing Over a <code>TreatmentResponseExperiment</code></a></li>
<li class="chapter" data-level="" data-path="module-2-data-preparation-for-pharmacogenomic-analysis.html"><a href="module-2-data-preparation-for-pharmacogenomic-analysis.html#quality-control-of-pharmacogenomic-datasets"><i class="fa fa-check"></i>Quality Control of Pharmacogenomic Datasets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><i class="fa fa-check"></i>Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i>Instructor(s) Name(s) and Contact Information</a></li>
<li class="chapter" data-level="" data-path="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#pharmacogx-package-overview"><i class="fa fa-check"></i>PharmacoGx Package Overview</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><i class="fa fa-check"></i>Module 4: Pharmacogenomics for Biomarker Discovery - Advanced Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#lab-3"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#overview-2"><i class="fa fa-check"></i>Overview</a></li>
<li class="chapter" data-level="" data-path="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html"><a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html#data-acquisition-and-preparation"><i class="fa fa-check"></i>Data Acquisition and Preparation</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="70" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">
<br><br>
<a href="https://computationalgenomics.ca/"><img src="img/sponsors/c3g.png" style="display:inline-block;" width=35% style="padding-bottom: 20px" alt="Canadian Centre for Computational Genomcis logo.">

<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li></center>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pharmacogenomics Data Analysis 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis" class="section level1 hasAnchor">
<h1>Module 3: Pharmacogenomics for Biomarker Discover - Basic Analysis<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#module-3-pharmacogenomics-for-biomarker-discover---basic-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="lecture-2" class="section level2 hasAnchor">
<h2>Lecture<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lecture-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<iframe width="640" height="340" src="https://www.youtube.com/embed/uiiYFyVEFBw?si=daES2wdCzbmuPGC2" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
<p><br></p>
<iframe src="https://drive.google.com/file/d/1qTd_ywKqpbAY6_UcCEVLjwoehZlieVvS/preview" width="640" height="480" allow="autoplay">
</iframe>
</div>
<div id="lab-2" class="section level2 hasAnchor">
<h2>Lab<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#lab-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="instructors-names-and-contact-information-1" class="section level3 unnumbered hasAnchor">
<h3>Instructor(s) Name(s) and Contact Information<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#instructors-names-and-contact-information-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Jermiah J. Joseph <a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a></li>
<li>Nikta Feizi <a href="mailto:nikta.feizi@uhn.ca" class="email">nikta.feizi@uhn.ca</a></li>
<li>Julia Nguyen <a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a></li>
</ul>
</div>
<div id="pharmacogx-package-overview" class="section level3 hasAnchor">
<h3>PharmacoGx Package Overview<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#pharmacogx-package-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>PharmacoGx:</strong> An R package for analyzing pharmacogenomic datasets.</p>
<p><strong>Key Functions:</strong></p>
<ul>
<li><p><strong>Install and Load PharmacoGx</strong></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb61-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb61-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb61-2" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;PharmacoGx&quot;</span>)</span>
<span id="cb61-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb61-3" tabindex="-1"></a><span class="fu">library</span>(PharmacoGx)</span></code></pre></div></li>
<li><p><strong>Download a PharmacoSet</strong></p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb62-1" tabindex="-1"></a><span class="fu">availablePSets</span>()</span>
<span id="cb62-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb62-2" tabindex="-1"></a>GDSC <span class="ot">&lt;-</span> <span class="fu">downloadPSet</span>(<span class="st">&quot;GDSC_2020(v2-8.2)&quot;</span>)</span>
<span id="cb62-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb62-3" tabindex="-1"></a><span class="co"># downloadPSet(): Download pharmacogenomic datasets.</span></span></code></pre></div></li>
<li><p><strong>Extract Drug Response Data</strong></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb63-1" tabindex="-1"></a>drug_response <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(GDSC, <span class="at">sensitivity.measure=</span><span class="st">&#39;aac_recomputed&#39;</span>)</span>
<span id="cb63-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb63-2" tabindex="-1"></a><span class="co"># summarizeSensitivityProfiles(): Summarize drug response data.</span></span></code></pre></div></li>
<li><p><strong>Extract Gene Expression Data</strong></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb64-1" tabindex="-1"></a>gene_expr <span class="ot">&lt;-</span> <span class="fu">summarizeMolecularProfiles</span>(GDSC, <span class="at">mDataType=</span><span class="st">&#39;rna&#39;</span>) </span>
<span id="cb64-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb64-2" tabindex="-1"></a>gene_expr_mtx <span class="ot">&lt;-</span> <span class="fu">assay</span>(gene_expr)</span></code></pre></div></li>
</ul>
<p><strong>Data Retrieval:</strong></p>
<ul>
<li><p><strong>availablePSets()</strong>: Lists available pharmacogenomic datasets.</p></li>
<li><p><strong>Downloading Multiple PharmacoSets:</strong></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb65-1" tabindex="-1"></a>PSet.list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;GDSC&quot;</span>, <span class="st">&quot;CCLE&quot;</span>, <span class="st">&quot;CTRP&quot;</span>), downloadPSet)</span>
<span id="cb65-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb65-2" tabindex="-1"></a><span class="fu">names</span>(PSet.list) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GDSC&quot;</span>, <span class="st">&quot;CCLE&quot;</span>, <span class="st">&quot;CTRP&quot;</span>)</span></code></pre></div></li>
<li><p><strong>Accessing Molecular Profiles:</strong></p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb66-1" tabindex="-1"></a>molecular.data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(PSet.list, <span class="cf">function</span>(pset) {</span>
<span id="cb66-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb66-2" tabindex="-1"></a>  <span class="fu">summarizeMolecularProfiles</span>(pset, <span class="at">mDataType =</span> <span class="st">&#39;rna&#39;</span>)</span>
<span id="cb66-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb66-3" tabindex="-1"></a>})</span></code></pre></div></li>
</ul>
<div id="introduction" class="section level4 hasAnchor">
<h4>Introduction<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In this workshop, we will explore data integration and comparative analysis using the <em><a href="https://bioconductor.org/packages/3.20/PharmacoGx">PharmacoGx</a></em> package, focusing on the <strong>GDSC</strong> and <strong>CCLE</strong> datasets. Specifically, we will cover drug sensitivity comparisons using AUC and IC50 measures and investigate correlations between gene expression profiles. This R Markdown file is designed to give participants hands-on experience with data exploration and interpretation.</p>
<div id="learning-objectives-2" class="section level5 hasAnchor">
<h5>Learning Objectives<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#learning-objectives-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li>Learn how to load and handle data from the <strong>GDSC</strong> and <strong>CCLE</strong> datasets.</li>
<li>Understand how to summarize drug sensitivity profiles and molecular data.</li>
<li>Perform correlations to assess concordance between cell line datasets.</li>
<li>Visualize and interpret the results of statistical tests.</li>
</ol>
</div>
</div>
<div id="exploring-gdsc-and-ccle-datasets" class="section level4 hasAnchor">
<h4>Exploring GDSC and CCLE Datasets<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#exploring-gdsc-and-ccle-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="load-gdsc-and-ccle-datasets" class="section level5 hasAnchor">
<h5>Load GDSC and CCLE Datasets<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#load-gdsc-and-ccle-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We will use the <strong>GDSCsmall</strong> and <strong>CCLEsmall</strong> sample datasets provided by <a href="https://bioconductor.org/packages/3.19/PharmacoGx"><em>PharmacoGx</em></a> to compare the two data sources.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;GDSCsmall&quot;</span>)</span>
<span id="cb67-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;CCLEsmall&quot;</span>)</span>
<span id="cb67-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-3" tabindex="-1"></a>GDSCsmall</span>
<span id="cb67-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-4" tabindex="-1"></a><span class="co">#&gt; &lt;PharmacoSet&gt;</span></span>
<span id="cb67-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-5" tabindex="-1"></a><span class="co">#&gt; Name: GDSC </span></span>
<span id="cb67-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-6" tabindex="-1"></a><span class="co">#&gt; Date Created: Mon Aug 24 15:18:23 2015 </span></span>
<span id="cb67-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-7" tabindex="-1"></a><span class="co">#&gt; Number of samples:  10 </span></span>
<span id="cb67-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-8" tabindex="-1"></a><span class="co">#&gt; Molecular profiles: </span></span>
<span id="cb67-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-9" tabindex="-1"></a><span class="co">#&gt; RNA :</span></span>
<span id="cb67-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-10" tabindex="-1"></a><span class="co">#&gt;   Dim: 300, 11 </span></span>
<span id="cb67-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-11" tabindex="-1"></a><span class="co">#&gt; rna2 :</span></span>
<span id="cb67-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-12" tabindex="-1"></a><span class="co">#&gt;   Dim: 300, 9 </span></span>
<span id="cb67-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-13" tabindex="-1"></a><span class="co">#&gt; mutation :</span></span>
<span id="cb67-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-14" tabindex="-1"></a><span class="co">#&gt;   Dim: 70, 10 </span></span>
<span id="cb67-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-15" tabindex="-1"></a><span class="co">#&gt; Treatment response: Drug pertubation:</span></span>
<span id="cb67-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-16" tabindex="-1"></a><span class="co">#&gt;    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span id="cb67-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-17" tabindex="-1"></a><span class="co">#&gt; Drug sensitivity:</span></span>
<span id="cb67-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-18" tabindex="-1"></a><span class="co">#&gt;    Number of Experiments:  972 </span></span>
<span id="cb67-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb67-19" tabindex="-1"></a><span class="co">#&gt;    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span></code></pre></div>
</div>
<div id="find-common-genes-between-gdsc-and-ccle" class="section level5 hasAnchor">
<h5>Find Common Genes Between GDSC and CCLE<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#find-common-genes-between-gdsc-and-ccle" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The first step in comparing datasets is to find shared features between them.
Here, we identify the genes that are present in both datasets.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb68-1" tabindex="-1"></a>commonGenes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">fNames</span>(GDSCsmall, <span class="st">&quot;rna&quot;</span>), <span class="fu">fNames</span>(CCLEsmall, <span class="st">&quot;rna&quot;</span>))</span>
<span id="cb68-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb68-2" tabindex="-1"></a><span class="fu">length</span>(commonGenes)</span>
<span id="cb68-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb68-3" tabindex="-1"></a><span class="co">#&gt; [1] 41</span></span></code></pre></div>
<p>There are 41 genes that are common between the RNA profiles of the GDSC and CCLE datasets.</p>
</div>
<div id="identify-common-cell-lines-and-drugs" class="section level5 hasAnchor">
<h5>Identify Common Cell Lines and Drugs<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#identify-common-cell-lines-and-drugs" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Next, we identify the common cell lines and drugs between <strong>GDSC</strong> and <strong>CCLE</strong>.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-1" tabindex="-1"></a>common <span class="ot">&lt;-</span> <span class="fu">intersectPSet</span>(</span>
<span id="cb69-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-2" tabindex="-1"></a>  <span class="fu">list</span>(<span class="st">&quot;CCLE&quot;</span> <span class="ot">=</span> CCLEsmall, <span class="st">&quot;GDSC&quot;</span> <span class="ot">=</span> GDSCsmall),</span>
<span id="cb69-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-3" tabindex="-1"></a>  <span class="at">intersectOn =</span> <span class="fu">c</span>(<span class="st">&quot;cell.lines&quot;</span>, <span class="st">&quot;drugs&quot;</span>),</span>
<span id="cb69-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-4" tabindex="-1"></a>  <span class="at">strictIntersect =</span> <span class="cn">TRUE</span></span>
<span id="cb69-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-5" tabindex="-1"></a>)</span>
<span id="cb69-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-6" tabindex="-1"></a><span class="co">#&gt; Intersecting large PSets may take a long time ...</span></span>
<span id="cb69-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-7" tabindex="-1"></a></span>
<span id="cb69-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-8" tabindex="-1"></a><span class="fu">cellNames</span>(common[[<span class="dv">1</span>]])</span>
<span id="cb69-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;22RV1&quot;    &quot;23132-87&quot; &quot;5637&quot;     &quot;639-V&quot;    &quot;647-V&quot;    &quot;697&quot;      &quot;769-P&quot;   </span></span>
<span id="cb69-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-10" tabindex="-1"></a><span class="co">#&gt; [8] &quot;786-0&quot;    &quot;8-MG-BA&quot;</span></span>
<span id="cb69-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-11" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">cellNames</span>(common[[<span class="dv">1</span>]]))</span>
<span id="cb69-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb69-12" tabindex="-1"></a><span class="co">#&gt; [1] 9</span></span></code></pre></div>
<p>There are 9 common samples between GDSC and CCLE. The <code>intersectPSet</code> function returns a list of the two PSets subsetted to include only the common cell lines.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-1" tabindex="-1"></a>common</span>
<span id="cb70-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-2" tabindex="-1"></a><span class="co">#&gt; $CCLE</span></span>
<span id="cb70-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-3" tabindex="-1"></a><span class="co">#&gt; &lt;PharmacoSet&gt;</span></span>
<span id="cb70-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-4" tabindex="-1"></a><span class="co">#&gt; Name: CCLE </span></span>
<span id="cb70-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-5" tabindex="-1"></a><span class="co">#&gt; Date Created: Fri Nov  6 14:00:53 2015 </span></span>
<span id="cb70-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-6" tabindex="-1"></a><span class="co">#&gt; Number of samples:  9 </span></span>
<span id="cb70-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-7" tabindex="-1"></a><span class="co">#&gt; Molecular profiles: </span></span>
<span id="cb70-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-8" tabindex="-1"></a><span class="co">#&gt; RNA :</span></span>
<span id="cb70-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-9" tabindex="-1"></a><span class="co">#&gt;   Dim: 50, 9 </span></span>
<span id="cb70-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-10" tabindex="-1"></a><span class="co">#&gt; RNAseq :</span></span>
<span id="cb70-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-11" tabindex="-1"></a><span class="co">#&gt;   Dim: 50, 9 </span></span>
<span id="cb70-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-12" tabindex="-1"></a><span class="co">#&gt; mutation :</span></span>
<span id="cb70-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-13" tabindex="-1"></a><span class="co">#&gt;   Dim: 1667, 9 </span></span>
<span id="cb70-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-14" tabindex="-1"></a><span class="co">#&gt; CNV :</span></span>
<span id="cb70-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-15" tabindex="-1"></a><span class="co">#&gt;   Dim: 50, 7 </span></span>
<span id="cb70-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-16" tabindex="-1"></a><span class="co">#&gt; Treatment response: Drug pertubation:</span></span>
<span id="cb70-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-17" tabindex="-1"></a><span class="co">#&gt;    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span id="cb70-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-18" tabindex="-1"></a><span class="co">#&gt; Drug sensitivity:</span></span>
<span id="cb70-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-19" tabindex="-1"></a><span class="co">#&gt;    Number of Experiments:  102 </span></span>
<span id="cb70-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-20" tabindex="-1"></a><span class="co">#&gt;    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span>
<span id="cb70-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb70-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-22" tabindex="-1"></a><span class="co">#&gt; $GDSC</span></span>
<span id="cb70-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-23" tabindex="-1"></a><span class="co">#&gt; &lt;PharmacoSet&gt;</span></span>
<span id="cb70-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-24" tabindex="-1"></a><span class="co">#&gt; Name: GDSC </span></span>
<span id="cb70-25"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-25" tabindex="-1"></a><span class="co">#&gt; Date Created: Mon Aug 24 15:18:23 2015 </span></span>
<span id="cb70-26"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-26" tabindex="-1"></a><span class="co">#&gt; Number of samples:  9 </span></span>
<span id="cb70-27"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-27" tabindex="-1"></a><span class="co">#&gt; Molecular profiles: </span></span>
<span id="cb70-28"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-28" tabindex="-1"></a><span class="co">#&gt; RNA :</span></span>
<span id="cb70-29"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-29" tabindex="-1"></a><span class="co">#&gt;   Dim: 300, 10 </span></span>
<span id="cb70-30"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-30" tabindex="-1"></a><span class="co">#&gt; rna2 :</span></span>
<span id="cb70-31"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-31" tabindex="-1"></a><span class="co">#&gt;   Dim: 300, 9 </span></span>
<span id="cb70-32"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-32" tabindex="-1"></a><span class="co">#&gt; mutation :</span></span>
<span id="cb70-33"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-33" tabindex="-1"></a><span class="co">#&gt;   Dim: 70, 9 </span></span>
<span id="cb70-34"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-34" tabindex="-1"></a><span class="co">#&gt; Treatment response: Drug pertubation:</span></span>
<span id="cb70-35"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-35" tabindex="-1"></a><span class="co">#&gt;    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span id="cb70-36"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-36" tabindex="-1"></a><span class="co">#&gt; Drug sensitivity:</span></span>
<span id="cb70-37"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-37" tabindex="-1"></a><span class="co">#&gt;    Number of Experiments:  79 </span></span>
<span id="cb70-38"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb70-38" tabindex="-1"></a><span class="co">#&gt;    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span></code></pre></div>
<p>Notice that most of the molecular profiles are subsetted to include the 9 common samples. Some odd cases include RNA from GDSC which has a technical replicate (hence 10 samples) and CNV from CCLE which is missing CNV profiles for two of the nine cell lines.</p>
</div>
<div id="summarize-drug-sensitivity-profiles" class="section level5 hasAnchor">
<h5>Summarize Drug Sensitivity Profiles<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#summarize-drug-sensitivity-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We summarize drug sensitivity profiles (AUC and IC50) for each dataset. The <code>summary.stat</code> parameter can be set to different statistical metrics such as <code>mean</code>, <code>median</code>, etc. Here, we use the <code>median</code> for summarizing the sensitivity.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-1" tabindex="-1"></a><span class="co"># Summary statistics for AUC</span></span>
<span id="cb71-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-2" tabindex="-1"></a>GDSC.auc <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb71-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-3" tabindex="-1"></a>  common<span class="sc">$</span>GDSC,</span>
<span id="cb71-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-4" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb71-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-5" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb71-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-6" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb71-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-7" tabindex="-1"></a>)</span>
<span id="cb71-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-8" tabindex="-1"></a>CCLE.auc <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb71-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-9" tabindex="-1"></a>  common<span class="sc">$</span>CCLE,</span>
<span id="cb71-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-10" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb71-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-11" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb71-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-12" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb71-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-13" tabindex="-1"></a>)</span>
<span id="cb71-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-14" tabindex="-1"></a></span>
<span id="cb71-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-15" tabindex="-1"></a><span class="co"># Summary statistics for IC50</span></span>
<span id="cb71-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-16" tabindex="-1"></a>GDSC.ic50 <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb71-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-17" tabindex="-1"></a>  common<span class="sc">$</span>GDSC,</span>
<span id="cb71-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-18" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;ic50_published&quot;</span>,</span>
<span id="cb71-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-19" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb71-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-20" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb71-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-21" tabindex="-1"></a>)</span>
<span id="cb71-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-22" tabindex="-1"></a>CCLE.ic50 <span class="ot">&lt;-</span> <span class="fu">summarizeSensitivityProfiles</span>(</span>
<span id="cb71-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-23" tabindex="-1"></a>  common<span class="sc">$</span>CCLE,</span>
<span id="cb71-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-24" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;ic50_published&quot;</span>,</span>
<span id="cb71-25"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-25" tabindex="-1"></a>  <span class="at">summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb71-26"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-26" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb71-27"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-27" tabindex="-1"></a>)</span>
<span id="cb71-28"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-28" tabindex="-1"></a>CCLE.auc <span class="sc">|&gt;</span> <span class="fu">head</span>()</span>
<span id="cb71-29"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-29" tabindex="-1"></a><span class="co">#&gt;               22RV1 23132-87      5637     639-V 647-V       697      769-P</span></span>
<span id="cb71-30"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-30" tabindex="-1"></a><span class="co">#&gt; PD-0325901 0.385000       NA 0.1708125 0.3088750    NA 0.2291000 0.23250000</span></span>
<span id="cb71-31"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-31" tabindex="-1"></a><span class="co">#&gt; 17-AAG     0.372460       NA 0.4828250 0.4877500    NA 0.3420875 0.40937500</span></span>
<span id="cb71-32"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-32" tabindex="-1"></a><span class="co">#&gt; Nilotinib  0.000000       NA 0.0072625 0.0710125    NA 0.1573375 0.00000000</span></span>
<span id="cb71-33"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-33" tabindex="-1"></a><span class="co">#&gt; PHA-665752 0.094375       NA 0.0000000 0.0194000    NA 0.1609750 0.09427143</span></span>
<span id="cb71-34"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-34" tabindex="-1"></a><span class="co">#&gt; lapatinib  0.036125       NA 0.2212250 0.0066000    NA 0.0453250 0.16475000</span></span>
<span id="cb71-35"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-35" tabindex="-1"></a><span class="co">#&gt; Nutlin-3   0.076925       NA 0.0000000 0.0666625    NA 0.2588125 0.08598571</span></span>
<span id="cb71-36"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-36" tabindex="-1"></a><span class="co">#&gt;                 786-0   8-MG-BA</span></span>
<span id="cb71-37"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-37" tabindex="-1"></a><span class="co">#&gt; PD-0325901 0.13862500 0.0758750</span></span>
<span id="cb71-38"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-38" tabindex="-1"></a><span class="co">#&gt; 17-AAG     0.44162500 0.4426500</span></span>
<span id="cb71-39"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-39" tabindex="-1"></a><span class="co">#&gt; Nilotinib  0.07501250        NA</span></span>
<span id="cb71-40"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-40" tabindex="-1"></a><span class="co">#&gt; PHA-665752 0.07408625 0.3962500</span></span>
<span id="cb71-41"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-41" tabindex="-1"></a><span class="co">#&gt; lapatinib  0.12987500 0.0547625</span></span>
<span id="cb71-42"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb71-42" tabindex="-1"></a><span class="co">#&gt; Nutlin-3   0.00000000 0.0655125</span></span></code></pre></div>
<p>We can visualize the distribution of drug sensitivity profiles to compare between the two psets.</p>
<p>First we quickly reformat the data into a long format to facilitate plotting. We use the <code>melt()</code> function from the <code>reshape2</code> package.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-1" tabindex="-1"></a>CCLE_toPlot <span class="ot">&lt;-</span> <span class="fu">melt</span>(CCLE.auc)</span>
<span id="cb72-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-2" tabindex="-1"></a><span class="fu">colnames</span>(CCLE_toPlot) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Drug&quot;</span>, <span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;AUC&quot;</span>)</span>
<span id="cb72-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-3" tabindex="-1"></a></span>
<span id="cb72-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-4" tabindex="-1"></a>CCLE_toPlot <span class="sc">|&gt;</span> <span class="fu">head</span>()</span>
<span id="cb72-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-5" tabindex="-1"></a><span class="co">#&gt;         Drug  Gene      AUC</span></span>
<span id="cb72-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-6" tabindex="-1"></a><span class="co">#&gt; 1 PD-0325901 22RV1 0.385000</span></span>
<span id="cb72-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-7" tabindex="-1"></a><span class="co">#&gt; 2     17-AAG 22RV1 0.372460</span></span>
<span id="cb72-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-8" tabindex="-1"></a><span class="co">#&gt; 3  Nilotinib 22RV1 0.000000</span></span>
<span id="cb72-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-9" tabindex="-1"></a><span class="co">#&gt; 4 PHA-665752 22RV1 0.094375</span></span>
<span id="cb72-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-10" tabindex="-1"></a><span class="co">#&gt; 5  lapatinib 22RV1 0.036125</span></span>
<span id="cb72-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb72-11" tabindex="-1"></a><span class="co">#&gt; 6   Nutlin-3 22RV1 0.076925</span></span></code></pre></div>
<p>Next, we can create a quick box plot to visualize the AUC distributions per drug.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb73-1" tabindex="-1"></a><span class="fu">ggplot</span>(CCLE_toPlot, <span class="fu">aes</span>(<span class="at">x =</span> Drug, <span class="at">y =</span> AUC)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb73-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb73-2" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="co">#rotate x-axis labels</span></span></code></pre></div>
<p><img src="_main_files/figure-html/plot_sensitivity-1.png" width="672" /></p>
<p>Let’s do this again but with the GDSC PSet so we can compare the two</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-1" tabindex="-1"></a><span class="co"># format the CCLE sensitivity data</span></span>
<span id="cb74-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-2" tabindex="-1"></a>GDSC_toPlot <span class="ot">&lt;-</span> <span class="fu">melt</span>(GDSC.auc)</span>
<span id="cb74-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-3" tabindex="-1"></a><span class="fu">colnames</span>(GDSC_toPlot) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Drug&quot;</span>, <span class="st">&quot;Gene&quot;</span>, <span class="st">&quot;AUC&quot;</span>)</span>
<span id="cb74-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-4" tabindex="-1"></a></span>
<span id="cb74-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-5" tabindex="-1"></a><span class="co"># merge the two pset dataframes</span></span>
<span id="cb74-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-6" tabindex="-1"></a>GDSC_toPlot<span class="sc">$</span>PSet <span class="ot">&lt;-</span> <span class="st">&quot;GDSC&quot;</span></span>
<span id="cb74-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-7" tabindex="-1"></a>CCLE_toPlot<span class="sc">$</span>PSet <span class="ot">&lt;-</span> <span class="st">&quot;CCLE&quot;</span></span>
<span id="cb74-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-8" tabindex="-1"></a>merge_toPlot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GDSC_toPlot, CCLE_toPlot)</span>
<span id="cb74-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-9" tabindex="-1"></a></span>
<span id="cb74-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-10" tabindex="-1"></a><span class="co"># plot to compare AUC distribution between PSets</span></span>
<span id="cb74-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-11" tabindex="-1"></a><span class="fu">ggplot</span>(merge_toPlot, <span class="fu">aes</span>(<span class="at">x =</span> Drug, <span class="at">y =</span> AUC)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> PSet)) <span class="sc">+</span></span>
<span id="cb74-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-12" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#624763&quot;</span>, <span class="st">&quot;#B1D3A3&quot;</span>)) <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb74-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb74-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="co">#rotate x-axis labels</span></span></code></pre></div>
<p><img src="_main_files/figure-html/plot_sensitivity_compare-1.png" width="672" /></p>
<p>Preclinical drug response data is fairly noisy, hence the variation we see between the two datasets. However, we do see some commonalities in distribution.</p>
</div>
<div id="summarize-gene-expression-profiles" class="section level5 hasAnchor">
<h5>Summarize Gene Expression Profiles<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#summarize-gene-expression-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We summarize gene expression data for the genes that are common between <strong>GDSC</strong> and <strong>CCLE</strong>.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-1" tabindex="-1"></a>GDSCexpression <span class="ot">&lt;-</span> <span class="fu">summarizeMolecularProfiles</span>(</span>
<span id="cb75-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-2" tabindex="-1"></a>  common<span class="sc">$</span>GDSC, <span class="fu">cellNames</span>(common<span class="sc">$</span>GDSC),</span>
<span id="cb75-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-3" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;rna&quot;</span>, <span class="at">features =</span> commonGenes, <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb75-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-4" tabindex="-1"></a>) <span class="sc">|&gt;</span> <span class="fu">assay</span>()</span>
<span id="cb75-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-5" tabindex="-1"></a>CCLEexpression <span class="ot">&lt;-</span> <span class="fu">summarizeMolecularProfiles</span>(</span>
<span id="cb75-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-6" tabindex="-1"></a>  common<span class="sc">$</span>CCLE, <span class="fu">cellNames</span>(common<span class="sc">$</span>CCLE),</span>
<span id="cb75-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-7" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;rna&quot;</span>, <span class="at">features =</span> commonGenes, <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb75-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-8" tabindex="-1"></a>) <span class="sc">|&gt;</span> <span class="fu">assay</span>() </span>
<span id="cb75-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-9" tabindex="-1"></a></span>
<span id="cb75-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-10" tabindex="-1"></a>GDSCexpression <span class="sc">|&gt;</span> <span class="fu">head</span>()</span>
<span id="cb75-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-11" tabindex="-1"></a><span class="co">#&gt;                     22RV1  23132-87      5637     639-V     647-V       697</span></span>
<span id="cb75-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-12" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003  5.890740  6.243345 10.018953  9.653157 10.039302  5.215355</span></span>
<span id="cb75-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-13" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005  4.448758  4.335738  4.108065  4.014940  4.250878  4.367339</span></span>
<span id="cb75-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-14" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419 10.427302 11.045576 11.611295 11.604278 11.248686 10.572231</span></span>
<span id="cb75-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-15" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457  5.905884  6.900556  5.684007  5.392485  5.748209  5.934834</span></span>
<span id="cb75-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-16" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460  6.010605  5.433439  5.729120  5.439058  6.453635  6.342021</span></span>
<span id="cb75-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-17" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938  4.410785  4.329849  4.397001  4.176387  4.275824  4.618786</span></span>
<span id="cb75-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-18" tabindex="-1"></a><span class="co">#&gt;                    769-P     786-0   8-MG-BA</span></span>
<span id="cb75-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-19" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003 8.615450  7.304893  7.994002</span></span>
<span id="cb75-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-20" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005 4.722970  4.235424  4.447925</span></span>
<span id="cb75-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-21" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419 9.927782 10.825907 10.759706</span></span>
<span id="cb75-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-22" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457 6.475836  5.087249  4.994176</span></span>
<span id="cb75-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-23" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460 5.999142  5.001772  6.073944</span></span>
<span id="cb75-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb75-24" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938 5.003250  4.443563  4.797094</span></span></code></pre></div>
<p>Here, you could do PCA on the profiles to quickly check the data. We will skip this step in the interest of time.</p>
</div>
</div>
<div id="correlation-analysis-between-gdsc-and-ccle" class="section level4 hasAnchor">
<h4>Correlation Analysis Between GDSC and CCLE<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#correlation-analysis-between-gdsc-and-ccle" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We perform correlation analysis to examine the relationship between gene expression, AUC, and IC50 measures across the <strong>GDSC</strong> and <strong>CCLE</strong> datasets using <strong>Spearman’s correlation</strong>.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-1" tabindex="-1"></a><span class="co"># get common cell line names</span></span>
<span id="cb76-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-2" tabindex="-1"></a>cc <span class="ot">&lt;-</span> <span class="fu">cellNames</span>(common[[<span class="dv">1</span>]])</span>
<span id="cb76-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-3" tabindex="-1"></a></span>
<span id="cb76-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-4" tabindex="-1"></a><span class="co"># correlation of gene expression across common cell lines</span></span>
<span id="cb76-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-5" tabindex="-1"></a>ge.cor <span class="ot">&lt;-</span> <span class="fu">sapply</span>(cc, <span class="cf">function</span>(x, d1, d2) {</span>
<span id="cb76-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-6" tabindex="-1"></a>  stats<span class="sc">::</span><span class="fu">cor</span>(</span>
<span id="cb76-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-7" tabindex="-1"></a>    d1[, x], d2[, x],</span>
<span id="cb76-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-8" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span></span>
<span id="cb76-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-9" tabindex="-1"></a>  )</span>
<span id="cb76-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-10" tabindex="-1"></a>}, <span class="at">d1 =</span> GDSCexpression, <span class="at">d2 =</span> CCLEexpression)</span>
<span id="cb76-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-11" tabindex="-1"></a></span>
<span id="cb76-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-12" tabindex="-1"></a>ge.cor</span>
<span id="cb76-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-13" tabindex="-1"></a><span class="co">#&gt;     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span id="cb76-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-14" tabindex="-1"></a><span class="co">#&gt; 0.8658537 0.8844948 0.8935540 0.9202091 0.9106272 0.8893728 0.8939024 0.8681185 </span></span>
<span id="cb76-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-15" tabindex="-1"></a><span class="co">#&gt;   8-MG-BA </span></span>
<span id="cb76-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb76-16" tabindex="-1"></a><span class="co">#&gt; 0.8470383</span></span></code></pre></div>
<p>We can see that the RNA-Seq expression values are highly correlated across the common cell lines.</p>
<p>Let’s take a look at the correlation of drug response, starting with IC50 values.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-1" tabindex="-1"></a><span class="co"># quick look at the CCLE IC50 data</span></span>
<span id="cb77-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-2" tabindex="-1"></a>CCLE.ic50 <span class="sc">|&gt;</span> <span class="fu">head</span>()</span>
<span id="cb77-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-3" tabindex="-1"></a><span class="co">#&gt;                22RV1 23132-87       5637     639-V 647-V       697     769-P</span></span>
<span id="cb77-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-4" tabindex="-1"></a><span class="co">#&gt; PD-0325901 8.0000000       NA 8.00000000 8.0000000    NA 8.0000000 1.4542717</span></span>
<span id="cb77-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-5" tabindex="-1"></a><span class="co">#&gt; 17-AAG     0.3297017       NA 0.07082279 0.1500945    NA 0.4225712 0.1517988</span></span>
<span id="cb77-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-6" tabindex="-1"></a><span class="co">#&gt; Nilotinib  8.0000000       NA 7.47535467 8.0000000    NA 1.9104344 8.0000000</span></span>
<span id="cb77-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-7" tabindex="-1"></a><span class="co">#&gt; PHA-665752 8.0000000       NA 8.00000000 8.0000000    NA 3.2754419 7.1504111</span></span>
<span id="cb77-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-8" tabindex="-1"></a><span class="co">#&gt; lapatinib  7.8473053       NA 2.30776763 8.0000000    NA 0.8494762 1.0574611</span></span>
<span id="cb77-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-9" tabindex="-1"></a><span class="co">#&gt; Nutlin-3   8.0000000       NA 8.00000000 8.0000000    NA 1.8905237 5.5913019</span></span>
<span id="cb77-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-10" tabindex="-1"></a><span class="co">#&gt;                786-0   8-MG-BA</span></span>
<span id="cb77-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-11" tabindex="-1"></a><span class="co">#&gt; PD-0325901 8.0000000 8.0000000</span></span>
<span id="cb77-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-12" tabindex="-1"></a><span class="co">#&gt; 17-AAG     0.2795302 0.2349526</span></span>
<span id="cb77-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-13" tabindex="-1"></a><span class="co">#&gt; Nilotinib  8.0000000        NA</span></span>
<span id="cb77-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-14" tabindex="-1"></a><span class="co">#&gt; PHA-665752 8.0000000 8.0000000</span></span>
<span id="cb77-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-15" tabindex="-1"></a><span class="co">#&gt; lapatinib  7.1780353 5.2266199</span></span>
<span id="cb77-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-16" tabindex="-1"></a><span class="co">#&gt; Nutlin-3   8.0000000 8.0000000</span></span>
<span id="cb77-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-17" tabindex="-1"></a></span>
<span id="cb77-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-18" tabindex="-1"></a><span class="co"># correlation of IC50 values across common drugs and cell lines</span></span>
<span id="cb77-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-19" tabindex="-1"></a>ic50.cor <span class="ot">&lt;-</span> <span class="fu">sapply</span>(cc, <span class="cf">function</span>(x, d1, d2) {</span>
<span id="cb77-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-20" tabindex="-1"></a>  stats<span class="sc">::</span><span class="fu">cor</span>(</span>
<span id="cb77-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-21" tabindex="-1"></a>    d1[, x], d2[, x],</span>
<span id="cb77-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-22" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span></span>
<span id="cb77-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-23" tabindex="-1"></a>  )</span>
<span id="cb77-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-24" tabindex="-1"></a>}, <span class="at">d1 =</span> GDSC.ic50, <span class="at">d2 =</span> CCLE.ic50)</span>
<span id="cb77-25"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-25" tabindex="-1"></a></span>
<span id="cb77-26"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-26" tabindex="-1"></a>ic50.cor</span>
<span id="cb77-27"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-27" tabindex="-1"></a><span class="co">#&gt;     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span id="cb77-28"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-28" tabindex="-1"></a><span class="co">#&gt; 0.4082483        NA 0.5713800 0.4082483        NA 0.6273461 0.6300619 0.6546537 </span></span>
<span id="cb77-29"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-29" tabindex="-1"></a><span class="co">#&gt;   8-MG-BA </span></span>
<span id="cb77-30"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb77-30" tabindex="-1"></a><span class="co">#&gt; 0.3664850</span></span></code></pre></div>
<p>Notice that we have some <code>NA</code> values in the original CCLE IC50 dataset and in the correlation results.</p>
<p>The <code>pairwise.complete.obs</code> argument used in the <code>cor()</code> function ensures that only complete observations (i.e. numeric values that are not <code>NA</code>) are included in the pairwise correlations.</p>
<p>Since <code>23132-87</code> and <code>647-V</code> are all <code>NA</code> values, no spearman correlation was computed.</p>
<p>We can quickly compute the correlation for AUC as well.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-1" tabindex="-1"></a><span class="co"># correlation of AUC values across common drugs and cell lines</span></span>
<span id="cb78-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-2" tabindex="-1"></a>auc.cor <span class="ot">&lt;-</span> <span class="fu">sapply</span>(cc, <span class="cf">function</span>(x, d1, d2) {</span>
<span id="cb78-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-3" tabindex="-1"></a>  stats<span class="sc">::</span><span class="fu">cor</span>(</span>
<span id="cb78-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-4" tabindex="-1"></a>    d1[, x], d2[, x],</span>
<span id="cb78-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-5" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span></span>
<span id="cb78-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-6" tabindex="-1"></a>  )</span>
<span id="cb78-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-7" tabindex="-1"></a>}, <span class="at">d1 =</span> GDSC.auc, <span class="at">d2 =</span> CCLE.auc)</span>
<span id="cb78-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-8" tabindex="-1"></a></span>
<span id="cb78-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-9" tabindex="-1"></a>auc.cor</span>
<span id="cb78-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-10" tabindex="-1"></a><span class="co">#&gt;     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span id="cb78-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-11" tabindex="-1"></a><span class="co">#&gt; 0.0000000        NA 0.7748062 0.8214286        NA 0.3750000 0.8571429 0.4638168 </span></span>
<span id="cb78-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-12" tabindex="-1"></a><span class="co">#&gt;   8-MG-BA </span></span>
<span id="cb78-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb78-13" tabindex="-1"></a><span class="co">#&gt; 0.5414447</span></span></code></pre></div>
</div>
<div id="statistical-comparison" class="section level4 hasAnchor">
<h4>Statistical Comparison<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#statistical-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We compare the correlations using <strong>Wilcoxon signed-rank tests</strong> to see if there are significant differences between the gene expression correlations and the drug sensitivity correlations.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-1" tabindex="-1"></a>w1 <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">wilcox.test</span>(</span>
<span id="cb79-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-2" tabindex="-1"></a>  <span class="at">x =</span> ge.cor, <span class="at">y =</span> auc.cor, <span class="at">conf.int =</span> <span class="cn">TRUE</span>, <span class="at">exact =</span> <span class="cn">FALSE</span></span>
<span id="cb79-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-3" tabindex="-1"></a>)</span>
<span id="cb79-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-4" tabindex="-1"></a>w2 <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">wilcox.test</span>(</span>
<span id="cb79-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-5" tabindex="-1"></a>  <span class="at">x =</span> ge.cor, <span class="at">y =</span> ic50.cor, <span class="at">conf.int =</span> <span class="cn">TRUE</span>, <span class="at">exact =</span> <span class="cn">FALSE</span></span>
<span id="cb79-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-6" tabindex="-1"></a>)</span>
<span id="cb79-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-7" tabindex="-1"></a></span>
<span id="cb79-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-8" tabindex="-1"></a>w1</span>
<span id="cb79-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb79-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-10" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb79-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb79-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-12" tabindex="-1"></a><span class="co">#&gt; data:  ge.cor and auc.cor</span></span>
<span id="cb79-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-13" tabindex="-1"></a><span class="co">#&gt; W = 62, p-value = 0.001496</span></span>
<span id="cb79-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-14" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb79-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-15" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb79-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-16" tabindex="-1"></a><span class="co">#&gt;  0.06309566 0.51857816</span></span>
<span id="cb79-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-17" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb79-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-18" tabindex="-1"></a><span class="co">#&gt; difference in location </span></span>
<span id="cb79-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-19" tabindex="-1"></a><span class="co">#&gt;               0.347967</span></span>
<span id="cb79-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-20" tabindex="-1"></a>w2</span>
<span id="cb79-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb79-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-22" tabindex="-1"></a><span class="co">#&gt;  Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb79-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb79-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-24" tabindex="-1"></a><span class="co">#&gt; data:  ge.cor and ic50.cor</span></span>
<span id="cb79-25"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-25" tabindex="-1"></a><span class="co">#&gt; W = 63, p-value = 0.001024</span></span>
<span id="cb79-26"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-26" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb79-27"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-27" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb79-28"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-28" tabindex="-1"></a><span class="co">#&gt;  0.2407670 0.4857055</span></span>
<span id="cb79-29"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-29" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb79-30"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-30" tabindex="-1"></a><span class="co">#&gt; difference in location </span></span>
<span id="cb79-31"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-31" tabindex="-1"></a><span class="co">#&gt;              0.3179657</span></span>
<span id="cb79-32"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-32" tabindex="-1"></a></span>
<span id="cb79-33"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-33" tabindex="-1"></a><span class="co"># Display p-values</span></span>
<span id="cb79-34"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-34" tabindex="-1"></a>ss <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;GE vs. AUC = %.1E</span><span class="sc">\n</span><span class="st">GE vs. IC50 = %.1E&quot;</span>, w1<span class="sc">$</span>p.value, w2<span class="sc">$</span>p.value)</span>
<span id="cb79-35"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-35" tabindex="-1"></a><span class="fu">cat</span>(ss)</span>
<span id="cb79-36"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-36" tabindex="-1"></a><span class="co">#&gt; GE vs. AUC = 1.5E-03</span></span>
<span id="cb79-37"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb79-37" tabindex="-1"></a><span class="co">#&gt; GE vs. IC50 = 1.0E-03</span></span></code></pre></div>
<div id="boxplot-visualization" class="section level5 hasAnchor">
<h5>Boxplot Visualization<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#boxplot-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The results are visualized using boxplots to compare the correlations across gene expression, AUC, and IC50.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-1" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">list</span>(<span class="st">&quot;GE&quot;</span> <span class="ot">=</span> ge.cor, <span class="st">&quot;AUC&quot;</span> <span class="ot">=</span> auc.cor, <span class="st">&quot;IC50&quot;</span> <span class="ot">=</span> ic50.cor),</span>
<span id="cb80-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-2" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Concordance between cell lines&quot;</span>,</span>
<span id="cb80-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-3" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="fu">expression</span>(R[s]),</span>
<span id="cb80-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-4" tabindex="-1"></a>  <span class="at">sub =</span> ss,</span>
<span id="cb80-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-5" tabindex="-1"></a>  <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb80-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-6" tabindex="-1"></a>  <span class="at">col =</span> <span class="st">&quot;lightgrey&quot;</span>,</span>
<span id="cb80-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-7" tabindex="-1"></a>  <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb80-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-8" tabindex="-1"></a>  <span class="at">border =</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb80-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb80-9" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/boxplot_results-1.png" width="672" /></p>
</div>
</div>
<div id="statistical-analysis-for-drug-response-associations" class="section level4 hasAnchor">
<h4>Statistical Analysis for Drug Response Associations<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#statistical-analysis-for-drug-response-associations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In this final section, we will be focusing on computing the association between a feature of interest and drug response in order to identify predictive biomarkers.</p>
<p>Let’s begin by selecting a feature and drug of interest to explore.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-1" tabindex="-1"></a>commonGenes <span class="sc">|&gt;</span> <span class="fu">head</span>()</span>
<span id="cb81-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot;</span></span>
<span id="cb81-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-3" tabindex="-1"></a><span class="co">#&gt; [5] &quot;ENSG00000000460&quot; &quot;ENSG00000000938&quot;</span></span>
<span id="cb81-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-4" tabindex="-1"></a>feature <span class="ot">&lt;-</span> <span class="st">&quot;ENSG00000000003&quot;</span></span>
<span id="cb81-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-5" tabindex="-1"></a></span>
<span id="cb81-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-6" tabindex="-1"></a>common<span class="sc">$</span>GDSC<span class="sc">@</span>treatment <span class="sc">|&gt;</span> <span class="fu">rownames</span>()</span>
<span id="cb81-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-7" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;PD-0325901&quot; &quot;17-AAG&quot;     &quot;Nilotinib&quot;  &quot;PHA-665752&quot; &quot;lapatinib&quot; </span></span>
<span id="cb81-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-8" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;Nutlin-3&quot;   &quot;AZD0530&quot;    &quot;Crizotinib&quot; &quot;Sorafenib&quot;  &quot;PD-0332991&quot;</span></span>
<span id="cb81-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-9" tabindex="-1"></a><span class="co">#&gt; [11] &quot;paclitaxel&quot; &quot;AZD6244&quot;    &quot;PLX4720&quot;    &quot;TAE684&quot;     &quot;Erlotinib&quot;</span></span>
<span id="cb81-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb81-10" tabindex="-1"></a>drug <span class="ot">&lt;-</span> <span class="st">&quot;PD-0325901&quot;</span></span></code></pre></div>
<p>Next, we need to extract the vector of gene expression for our chosen feature and drug response for our chosen drug. Let’s do this for the GDSC dataset.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-1" tabindex="-1"></a>Gene <span class="ot">&lt;-</span> GDSCexpression[feature,]</span>
<span id="cb82-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-2" tabindex="-1"></a>Drug <span class="ot">&lt;-</span> GDSC.auc[drug,]</span>
<span id="cb82-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-3" tabindex="-1"></a></span>
<span id="cb82-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-4" tabindex="-1"></a>Gene</span>
<span id="cb82-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-5" tabindex="-1"></a><span class="co">#&gt;     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span id="cb82-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-6" tabindex="-1"></a><span class="co">#&gt;  5.890740  6.243345 10.018953  9.653157 10.039302  5.215355  8.615450  7.304893 </span></span>
<span id="cb82-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-7" tabindex="-1"></a><span class="co">#&gt;   8-MG-BA </span></span>
<span id="cb82-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-8" tabindex="-1"></a><span class="co">#&gt;  7.994002</span></span>
<span id="cb82-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-9" tabindex="-1"></a>Drug</span>
<span id="cb82-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-10" tabindex="-1"></a><span class="co">#&gt;    22RV1 23132-87     5637    639-V    647-V      697    769-P    786-0 </span></span>
<span id="cb82-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-11" tabindex="-1"></a><span class="co">#&gt; 0.029649 0.122911 0.007242 0.279261 0.100351 0.049681 0.179831 0.150231 </span></span>
<span id="cb82-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-12" tabindex="-1"></a><span class="co">#&gt;  8-MG-BA </span></span>
<span id="cb82-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb82-13" tabindex="-1"></a><span class="co">#&gt; 0.034754</span></span></code></pre></div>
<p>Notice that the PSet has already kept the samples ordered across the various profiles.</p>
<p>Before we compute the associations, we can quickly plot the two variables to visualize their relationship.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb83-1" tabindex="-1"></a><span class="co"># create datafarme to plot</span></span>
<span id="cb83-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb83-2" tabindex="-1"></a>toPlot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Gene =</span> Gene, <span class="at">Drug =</span> Drug)</span>
<span id="cb83-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb83-3" tabindex="-1"></a></span>
<span id="cb83-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb83-4" tabindex="-1"></a><span class="fu">ggplot</span>(toPlot, <span class="fu">aes</span>(<span class="at">x =</span> Gene, <span class="at">y =</span> Drug)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb83-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb83-5" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Gene (expr)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Drug (AUC)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_association-1.png" width="672" /></p>
<p>From the scatter plot, we see a weak positive correlation. Let’s see if this is reflected in our association analyses.</p>
<div id="concordance-index" class="section level5 hasAnchor">
<h5>Concordance Index<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#concordance-index" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Fist, we will compute the association between the gene expression data and drug response. We will use the <code>concordance.index</code> function from the <code>survcomp</code> package.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-1" tabindex="-1"></a>ci <span class="ot">&lt;-</span> survcomp<span class="sc">::</span><span class="fu">concordance.index</span>(</span>
<span id="cb84-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-2" tabindex="-1"></a>  <span class="fu">as.numeric</span>(Drug),                   <span class="co"># drug vector</span></span>
<span id="cb84-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-3" tabindex="-1"></a>  <span class="at">surv.time =</span> <span class="fu">as.numeric</span>(Gene),       <span class="co"># gene vector</span></span>
<span id="cb84-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-4" tabindex="-1"></a>  <span class="at">surv.event =</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">length</span>(Gene)),   </span>
<span id="cb84-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-5" tabindex="-1"></a>  <span class="at">outx =</span> <span class="cn">TRUE</span>, <span class="at">method=</span><span class="st">&quot;noether&quot;</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb84-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-6" tabindex="-1"></a>)</span>
<span id="cb84-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-7" tabindex="-1"></a></span>
<span id="cb84-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Concordance Index:&quot;</span>, ci<span class="sc">$</span>c.index, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb84-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-9" tabindex="-1"></a>    <span class="st">&quot;P-value:&quot;</span>, ci<span class="sc">$</span>p.value, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb84-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-10" tabindex="-1"></a>    <span class="st">&quot;Standard Error:&quot;</span>, ci<span class="sc">$</span>se, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb84-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-11" tabindex="-1"></a>    <span class="st">&quot;Upper CI:&quot;</span>, ci<span class="sc">$</span>upper,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb84-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-12" tabindex="-1"></a>    <span class="st">&quot;Lower CI:&quot;</span>, ci<span class="sc">$</span>lower)</span>
<span id="cb84-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-13" tabindex="-1"></a><span class="co">#&gt; Concordance Index: 0.4166667 </span></span>
<span id="cb84-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-14" tabindex="-1"></a><span class="co">#&gt;  P-value: 0.07592696 </span></span>
<span id="cb84-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-15" tabindex="-1"></a><span class="co">#&gt;  Standard Error: 0.04695301 </span></span>
<span id="cb84-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-16" tabindex="-1"></a><span class="co">#&gt;  Upper CI: 0.5086929 </span></span>
<span id="cb84-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb84-17" tabindex="-1"></a><span class="co">#&gt;  Lower CI: 0.3246404</span></span></code></pre></div>
<p>The concordance index tells us there is a weak inverse association between the gene expression and the drug response.</p>
</div>
<div id="pearsons-correlation-coefficient" class="section level5 hasAnchor">
<h5>Pearson’s Correlation Coefficient<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#pearsons-correlation-coefficient" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Using the same feature and drug, let’s compute the association again but this time using Pearson’s correlation.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-1" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb85-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-2" tabindex="-1"></a>  <span class="at">x =</span> Gene,</span>
<span id="cb85-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-3" tabindex="-1"></a>  <span class="at">y =</span> Drug,</span>
<span id="cb85-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-4" tabindex="-1"></a>  <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>,</span>
<span id="cb85-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-5" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span></span>
<span id="cb85-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-6" tabindex="-1"></a>)</span>
<span id="cb85-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-7" tabindex="-1"></a></span>
<span id="cb85-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Pearson&#39;s correlation:&quot;</span>, ps<span class="sc">$</span>estimate, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb85-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-9" tabindex="-1"></a>    <span class="st">&quot;P-value:&quot;</span>, ci<span class="sc">$</span>p.value, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb85-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-10" tabindex="-1"></a>    <span class="st">&quot;Confidence Interval:&quot;</span>, ps<span class="sc">$</span>conf.int)</span>
<span id="cb85-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-11" tabindex="-1"></a><span class="co">#&gt; Pearson&#39;s correlation: 0.2937677 </span></span>
<span id="cb85-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-12" tabindex="-1"></a><span class="co">#&gt;  P-value: 0.07592696 </span></span>
<span id="cb85-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb85-13" tabindex="-1"></a><span class="co">#&gt;  Confidence Interval: -0.4601228 0.8015157</span></span></code></pre></div>
<p>The Pearson’s correlation tells us there is a (very) weak positive linear relationship between the gene expression and the drug response.</p>
<p>Notice the slight difference in interpretation between the concordance index and the Pearson’s correlation.</p>
</div>
<div id="drug-sensitivity-signatures-pharmacogx" class="section level5 hasAnchor">
<h5>Drug Sensitivity Signatures (PharmacoGx)<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#drug-sensitivity-signatures-pharmacogx" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The last method we will explore is a built-in function from the <code>PharmacoGx</code> package called <code>drugSensitivitySig</code>. This function takes a <code>PharmacoSet</code> along with a list of drugs and features, then computes the association between the feature expression and drug response for each pair.</p>
<p>Let’s first select the first 3 genes and drugs as our features &amp; drugs of interest:</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-1" tabindex="-1"></a>features <span class="ot">&lt;-</span> commonGenes[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb86-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-2" tabindex="-1"></a>drugs <span class="ot">&lt;-</span> <span class="fu">rownames</span>(common<span class="sc">$</span>GDSC<span class="sc">@</span>treatment)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb86-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-3" tabindex="-1"></a></span>
<span id="cb86-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-4" tabindex="-1"></a>features</span>
<span id="cb86-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot;</span></span>
<span id="cb86-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-6" tabindex="-1"></a>drugs</span>
<span id="cb86-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb86-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;PD-0325901&quot; &quot;17-AAG&quot;     &quot;Nilotinib&quot;</span></span></code></pre></div>
<p>Now we can calculate the drug sensitivity signature using the vector of genes and drugs for the GDSC dataset.</p>
<div class="callout-blue">
<span class="callout-icon"><i class="fas fa-circle-info" aria-hidden="true"></i></span>
<div class="callout-content">
<div class="callout-title-static" style="margin-bottom: 0.5em;">
<p>Note</p>
</div>
<p>This gene-drug signature is based on univariable analysis.</p>
</div>
</div>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-1" tabindex="-1"></a><span class="co"># gene expression signature</span></span>
<span id="cb87-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-2" tabindex="-1"></a>sig.rna <span class="ot">&lt;-</span> <span class="fu">drugSensitivitySig</span>(</span>
<span id="cb87-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-3" tabindex="-1"></a>  <span class="at">object =</span> GDSCsmall,</span>
<span id="cb87-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-4" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb87-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-5" tabindex="-1"></a>  <span class="at">drugs =</span> drugs,</span>
<span id="cb87-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-6" tabindex="-1"></a>  <span class="at">features =</span> features,</span>
<span id="cb87-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-7" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb87-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-8" tabindex="-1"></a>  <span class="at">molecular.summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb87-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-9" tabindex="-1"></a>  <span class="at">sensitivity.summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb87-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-10" tabindex="-1"></a>  <span class="at">modeling.method =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb87-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-11" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb87-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-12" tabindex="-1"></a>)</span>
<span id="cb87-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-13" tabindex="-1"></a>sig.rna<span class="sc">@</span>.Data[,,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>)]</span>
<span id="cb87-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-14" tabindex="-1"></a><span class="co">#&gt; , , estimate</span></span>
<span id="cb87-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb87-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-16" tabindex="-1"></a><span class="co">#&gt;                  PD-0325901     17-AAG  Nilotinib</span></span>
<span id="cb87-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-17" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003 -0.18840373 -0.2369481  0.8554590</span></span>
<span id="cb87-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-18" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005 -0.14409315 -0.4756331  0.7875267</span></span>
<span id="cb87-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-19" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419 -0.02890968  0.5466780 -0.8239237</span></span>
<span id="cb87-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb87-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-21" tabindex="-1"></a><span class="co">#&gt; , , pvalue</span></span>
<span id="cb87-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb87-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-23" tabindex="-1"></a><span class="co">#&gt;                 PD-0325901    17-AAG Nilotinib</span></span>
<span id="cb87-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-24" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003  0.8793373 0.8477058 0.3465502</span></span>
<span id="cb87-25"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-25" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005  0.9079470 0.6844380 0.4227225</span></span>
<span id="cb87-26"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb87-26" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419  0.9815930 0.6317843 0.3835609</span></span></code></pre></div>
<p>The output is a 3D array of genes x drugs x metric. Outputted are the <code>estimate</code> and <code>pvalue</code>. Feel free to explore the other array components in <code>sig.rna@.Data</code>.</p>
<p>This function is a powerful method for investigating the univariate relationship between multiple molecular features and drugs.</p>
<p>Let’s do the same for mutations, we’ll do a quick example using one gene and drug from the CCLE dataset.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-1" tabindex="-1"></a><span class="co"># mutation signature</span></span>
<span id="cb88-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-2" tabindex="-1"></a>sig.mut <span class="ot">&lt;-</span> <span class="fu">drugSensitivitySig</span>(</span>
<span id="cb88-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-3" tabindex="-1"></a>  <span class="at">object =</span> CCLEsmall,</span>
<span id="cb88-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-4" tabindex="-1"></a>  <span class="at">mDataType =</span> <span class="st">&quot;mutation&quot;</span>,</span>
<span id="cb88-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-5" tabindex="-1"></a>  <span class="at">drugs =</span> <span class="st">&quot;PD-0325901&quot;</span>,</span>
<span id="cb88-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-6" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">&quot;BRAF&quot;</span>,</span>
<span id="cb88-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-7" tabindex="-1"></a>  <span class="at">sensitivity.measure =</span> <span class="st">&quot;auc_published&quot;</span>,</span>
<span id="cb88-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-8" tabindex="-1"></a>  <span class="at">molecular.summary.stat =</span> <span class="st">&quot;and&quot;</span>,</span>
<span id="cb88-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-9" tabindex="-1"></a>  <span class="at">sensitivity.summary.stat =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb88-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-10" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb88-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-11" tabindex="-1"></a>)</span>
<span id="cb88-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-12" tabindex="-1"></a>sig.mut<span class="sc">@</span>.Data[,,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>)]</span>
<span id="cb88-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-13" tabindex="-1"></a><span class="co">#&gt;     estimate       pvalue </span></span>
<span id="cb88-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb88-14" tabindex="-1"></a><span class="co">#&gt; 8.251857e-01 1.612881e-09</span></span></code></pre></div>
</div>
</div>
<div id="univariate-vs-multivariate-analysis" class="section level4 hasAnchor">
<h4>Univariate vs Multivariate Analysis<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#univariate-vs-multivariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For the final section of this lab, we will explore univariate vs multivariate linear modeling. Let’s start with a simple univariate linear model using the same Gene and Drug vector from earlier.</p>
<div id="univariate-analysis" class="section level5 hasAnchor">
<h5>Univariate Analysis<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#univariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Recall <code>Gene</code> is the RNA expression vector of <code>ENSG00000000005</code> and <code>Drug</code> is the drug response vector (AUC) for <code>PD-0325901</code> in the GDSC dataset.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-1" tabindex="-1"></a><span class="fu">lm</span>(Drug <span class="sc">~</span> Gene) <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb89-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb89-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-3" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb89-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-4" tabindex="-1"></a><span class="co">#&gt; lm(formula = Drug ~ Gene)</span></span>
<span id="cb89-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb89-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-6" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb89-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-7" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb89-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-8" tabindex="-1"></a><span class="co">#&gt; -0.12867 -0.04835 -0.01884  0.05239  0.14848 </span></span>
<span id="cb89-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb89-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-10" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb89-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-11" tabindex="-1"></a><span class="co">#&gt;              Estimate Std. Error t value Pr(&gt;|t|)</span></span>
<span id="cb89-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-12" tabindex="-1"></a><span class="co">#&gt; (Intercept) -0.004639   0.139300  -0.033    0.974</span></span>
<span id="cb89-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-13" tabindex="-1"></a><span class="co">#&gt; Gene         0.014028   0.017253   0.813    0.443</span></span>
<span id="cb89-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb89-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-15" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 0.08965 on 7 degrees of freedom</span></span>
<span id="cb89-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-16" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.0863, Adjusted R-squared:  -0.04423 </span></span>
<span id="cb89-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb89-17" tabindex="-1"></a><span class="co">#&gt; F-statistic: 0.6612 on 1 and 7 DF,  p-value: 0.4429</span></span></code></pre></div>
<p>Let’s break down this output:</p>
<p><em>Residuals</em>: the difference between the predicted y (drug response) values and the
actual y values. A larger range suggests the model may not be a good fit of the data.</p>
<p><em>Coefficients</em>: the parameters of the model.</p>
<ul>
<li><code>Estimate</code> can be considered the effect size. Specifically, it is the one-unit change in the y variable (drug response) per one-unit change in the x variable (gene expression). The estimate from our drug~gene association is <code>0.014028</code>.</li>
<li><code>Pr(&gt;|t|)</code> is the corresponding p-value for each estimate. The p-value from our drug~gene association test is <code>0.443</code>.</li>
</ul>
<div class="callout-blue">
<span class="callout-icon"><i class="fas fa-circle-info" aria-hidden="true"></i></span>
<div class="callout-content">
<div class="callout-title-static" style="margin-bottom: 0.5em;">
<p>Note</p>
</div>
<p>If any of the coefficients had met the p-value &lt; 0.05 significance threshold, you would have seen <code>*</code> beside the estimate (unfortunately, this association does not meet this threshold).</p>
</div>
</div>
<p>Let’s see if we can make a multi-variate model with better performance.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-1" tabindex="-1"></a><span class="co"># get another gene vector</span></span>
<span id="cb90-2"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-2" tabindex="-1"></a>commonGenes[<span class="dv">2</span>]</span>
<span id="cb90-3"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ENSG00000000005&quot;</span></span>
<span id="cb90-4"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-4" tabindex="-1"></a>Gene2 <span class="ot">&lt;-</span> GDSCexpression[commonGenes[<span class="dv">2</span>],]</span>
<span id="cb90-5"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-5" tabindex="-1"></a></span>
<span id="cb90-6"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-6" tabindex="-1"></a><span class="co"># multi-variate model</span></span>
<span id="cb90-7"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-7" tabindex="-1"></a><span class="fu">lm</span>(Drug <span class="sc">~</span> Gene <span class="sc">+</span> Gene2) <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb90-8"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb90-9"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-9" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb90-10"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-10" tabindex="-1"></a><span class="co">#&gt; lm(formula = Drug ~ Gene + Gene2)</span></span>
<span id="cb90-11"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb90-12"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-12" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb90-13"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-13" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb90-14"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-14" tabindex="-1"></a><span class="co">#&gt; -0.13292 -0.04715 -0.02147  0.04840  0.14047 </span></span>
<span id="cb90-15"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb90-16"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-16" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb90-17"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-17" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)</span></span>
<span id="cb90-18"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-18" tabindex="-1"></a><span class="co">#&gt; (Intercept)  0.15618    0.84445   0.185    0.859</span></span>
<span id="cb90-19"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-19" tabindex="-1"></a><span class="co">#&gt; Gene         0.01251    0.02017   0.620    0.558</span></span>
<span id="cb90-20"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-20" tabindex="-1"></a><span class="co">#&gt; Gene2       -0.03441    0.17780  -0.194    0.853</span></span>
<span id="cb90-21"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb90-22"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-22" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 0.09653 on 6 degrees of freedom</span></span>
<span id="cb90-23"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-23" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.09197,    Adjusted R-squared:  -0.2107 </span></span>
<span id="cb90-24"><a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#cb90-24" tabindex="-1"></a><span class="co">#&gt; F-statistic: 0.3038 on 2 and 6 DF,  p-value: 0.7487</span></span></code></pre></div>
<p>The second gene feature we are adding to our model is <code>ENSG00000000005</code>.</p>
<p>Take a look at the output, comparing it to the previous model we ran. Do you think that adding an additional gene has improved model performance?</p>
</div>
</div>
<div id="individual-practice-1" class="section level4 hasAnchor">
<h4>Individual Practice<a href="module-3-pharmacogenomics-for-biomarker-discover---basic-analysis.html#individual-practice-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For the remainder of the lab, please explore the associations between other features and drug response. Try using the CCLE pset too!</p>
<div class="callout-green">
<span class="callout-icon"><i class="fas fa-circle-check" aria-hidden="true"></i></span>
<div class="callout-content">
<div class="callout-title-static" style="margin-bottom: 0.5em;">
<p>Lab Completed!</p>
</div>
<p>Congratulations! You have completed Lab 3!</p>
</div>
</div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-2-data-preparation-for-pharmacogenomic-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-4-pharmacogenomics-for-biomarker-discovery---advanced-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/bioinformaticsdotca/PGX_2024/blob/main/030-module-3.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
